src/main.rs

```diff
@@ -90,31 +95,62 @@ impl HanoiError {
 /// ## Returns
 ///
 ///    * `Action`: if the input was well formed
-///    * `Hanoi::UnknownCommand`: otherwise
+///    * `HanoiError::UnknownCommand`: otherwise
 fn parse_action(input: &str) -> Result<Action,HanoiError> {
-    unimplemented!()
+	//string matching too hard; just enumerate all the things
+    match input {
+		"q" 	=> Ok(Action::Quit),
+		"lc"	=> Ok(Action::Move(Move::new(Peg::Left, Peg::Center))),
+		"lr"	=> Ok(Action::Move(Move::new(Peg::Left, Peg::Right))),
+		"cl"	=> Ok(Action::Move(Move::new(Peg::Center, Peg::Left))),
+		"cr"	=> Ok(Action::Move(Move::new(Peg::Center, Peg::Right))),
+		"rl"	=> Ok(Action::Move(Move::new(Peg::Right, Peg::Left))),
+		"rc"	=> Ok(Action::Move(Move::new(Peg::Right, Peg::Center))),
+		_		=> Err(HanoiError::UnknownCommand),
+	}
 }
 
 impl State {
 
     /// Creates a Towers of Hanoi game with `disks` disks in a single tower
     fn new(disks: u8) -> State {
-        unimplemented!()
+		//need a mut vector so that we can do nice operations on it
+		//it's more efficient to just set up the iterator nicely, but it's just not worth doing
+        let mut starting_vector: Vec<Disk> = Vec::new();
+		for size in 0..disks {
+			starting_vector.push(Disk(disks-size))
+		}
```

Comment:
You can turn a range like `0..disks` (or any iterator) into a vector using `Iterator::collect`. See the sample solution!